lib/model/model.go

```diff
@@ -2418,6 +2444,12 @@ func (m *model) CommitConfiguration(from, to config.Configuration) bool {
 		if !ok {
 			// The folder was removed.
 			m.removeFolder(fromCfg)
+			// Forget pending folder device associations as well, assuming the
+			// folder is no longer of interest at all (but might become
+			// pending again).
+			for _, dev := range from.Devices {
+				forgetPending.MarkFolder(folderID, []protocol.DeviceID{dev.DeviceID}, to.MyID)
+			}
```

Comment:
Shouldn't it be enough to mark the entire device for removal instead of all the folders individually, which then drops all the pending folder entries for said device?

Reply:
I don't understand what you mean.  This place is about removed **folders**.  Which **device** would you want to clean from the PendingFolders then?

Note that technically this is more than needed here (which the comment tries to explain): The folder was only present in the old configuration.  This loop iterates over all devices known then, not only those which had it shared to already.  That set of devices is exhaustive, since we only ever accept ClusterConfigs from already known (configured) devices.  For some of them a PendingFolder record could exist and is still technically valid.  But we assume that a user locally no longer interested in the folder would not want lingering (possibly stale) offers of it from other devices.  If that assumption does not hold, the offers will come back at the next ClusterConfig exchange.

That's the reasoning why I didn't include any "wildcard" semantics for the **folder ID** in forgetPending.

Reply:
I made a mistake in my reasoning, I thought for a dropped pending device we anyway also need to drop all it's folders, but that's obviously not the case.

Reply:
Is this logic good to go then?  Do you agree that these extra pending folder entries should be removed although they are technically not invalidated?

I think that is actually a functional change, because previously noone removed `<pendingFolder>` entries under `<device>` if the corresponding `<folder>` was removed from the XML config.

Reply:
Oh and please do indicate when it's okay for me to resolve a discussion.  From your previous comment that was not quite clear.

Reply:
Yes I agree the logic is good. And I actually expected this PR to potentially change behaviour, because the existing behaviour is clearly lacking in some regards (we simply don't handle a few scenarios).  
(From my side consensus is reached now: If you do have something further to comment on this topic, do so and don't resolve, if this is also finished from your side, just resolve without commenting).